src/gt4py/next/iterator/tracing.py

```diff
@@ -172,6 +174,7 @@ def trace_function_call(fun, *, args=None):
     body = fun(*list(args))
 
     if isinstance(body, tuple):
+        # TODO(tehrengruber): This fails for nested tuples.
         return _f("make_tuple", *tuple(make_node(b) for b in body))
     else:
         return make_node(body) if body is not None else None
```

Comment:
Actually why does this not work?

```suggestion
    return make_node(body) if body is not None else None
```

Reply:
Fixed in #1317